### ISSUE TITLE
Routine name always quoted

### ISSUE THREAD

**Issue by flemingadam** [jOOQ/jOOQ#9813]
### Expected behavior and actual behavior:

Extensions of `AbstractRoutine` should use the rendering preferences defined in the configuration.

In jOOQ 3.11.9 using the POSTGRES dialact, `RenderNameStyle.AS_IS`, and `RenderKeywordStyle.AS_IS`, a routine with the name `my_function` would render as `my_function(a,b,c)` .  In jOOQ 3.12.4, it renders as `"my_function"(a,b,c)`.  With some PostgreSQL functions, this is invalid syntax.

### Steps to reproduce the problem (if possible, create an MCVE: https://github.com/jOOQ/jOOQ-mcve):

The following text file is a single class that shows the issue:
[jooq-mcve.txt](https://github.com/jOOQ/jOOQ/files/4171611/jooq-mcve.txt)

1. Run the MVCE using jOOQ 3.11.9, which renders the routine as

```sql
select my_function(
  a := true, 
  b := false
)
```

2. Run the MVCE using jOOQ 3.12.4, which renders the routine as
jOOQ 3.12.4 renders as:
```sql
select "my_function"(
  a := true, 
  b := false
)
```
jOOQ 3.12.4 adds `RenderQuotedNames` options, and adding
```java
settings.setRenderQuotedNames(RenderQuotedNames.NEVER);
```
does not change anything.

### Suspected Cause

When rendering the function name, the `AbstractContext.quote()` method returns `false` in 3.11.9 and `true` in 3.12.4.  In jOOQ 3.11.9, the `AbstractContext` class has the following line in the constructor that sets the `quote` variable based on the configuration:
```java
this.quote = settings().getRenderNameStyle() == RenderNameStyle.QUOTED;
```
In jOOQ 3.12.4, the constructor has been refactored and the `quote` variable does not appear to be set.

### Versions:

- jOOQ: 3.12.4
- Java: 11.0.2
- Database (include vendor): PostgreSQL
- OS: Windows
- JDBC Driver (include name if inofficial driver): N/A

**Comment by lukaseder** [comment]
Thanks a lot for your report. We'll investigate this soon

**Comment by lukaseder** [comment]
Related issue: https://stackoverflow.com/q/60257963/521799

**Comment by lukaseder** [comment]
I can still reproduce this with 3.13.2, but no longer with 3.14.0-SNAPSHOT. There have been some improvements in this area, recently. It seems that this got fixed already.

**Comment by lukaseder** [comment]
This has been fixed via a separate issue: https://github.com/jOOQ/jOOQ/issues/10269
Also fixed in 3.13.3-SNAPSHOT: https://github.com/jOOQ/jOOQ/issues/10270

Closing as duplicate of https://github.com/jOOQ/jOOQ/issues/10269
